presentation/src/main/java/com/yapp/growth/presentation/ui/main/create/timerange/TimeRangeViewModel.kt

```diff
@@ -0,0 +1,103 @@
+        chosenTheme?.let {
+            updateState {
+                copy(
+                    chosenTheme = PlanThemeType.valueOf(chosenTheme),
+                    title = if (title.isNullOrBlank()) "" else title,
+                    place = if (place.isNullOrBlank()) "" else place,
+                    dates = if (dates.isNullOrBlank()) "" else dates,
+                )
+            }
+        }
+    }
+
+    override fun handleEvents(event: TimeRangeEvent) {
+        when (event) {
+            is TimeRangeEvent.OnClickExitButton -> sendEffect({ TimeRangeSideEffect.ExitCreateScreen })
+            is TimeRangeEvent.OnClickNextButton -> sendEffect({ TimeRangeSideEffect.NavigateToNextScreen })
+            is TimeRangeEvent.OnClickBackButton -> sendEffect({ TimeRangeSideEffect.NavigateToPreviousScreen })
+            is TimeRangeEvent.OnSelectedHourChanged -> {
+                updateHourState(viewState.value.dialogState, event.hour)
+                sendEffect({ TimeRangeSideEffect.HideBottomSheet })
+                checkRangeError(viewState.value.startHour, viewState.value.endHour)
+            }
+            is TimeRangeEvent.OnStartHourClicked -> {
+                updateState { copy(dialogState = TimeRangeViewState.DialogState.START_HOUR) }
+                sendEffect({ TimeRangeSideEffect.ShowBottomSheet })
+            }
+            is TimeRangeEvent.OnEndHourClicked -> {
+                updateState { copy(dialogState = TimeRangeViewState.DialogState.END_HOUR) }
+                sendEffect({ TimeRangeSideEffect.ShowBottomSheet })
+            }
+        }
+    }
+
+    private fun updateHourState(dialogState: TimeRangeViewState.DialogState, hour: Int) {
+        when (dialogState) {
+            TimeRangeViewState.DialogState.START_HOUR -> {
+                updateState {
+                    copy(
+                        startHour = hour,
+                        dialogState = TimeRangeViewState.DialogState.NONE
+                    )
+                }
+            }
+            TimeRangeViewState.DialogState.END_HOUR -> updateState {
+                copy(
+                    endHour = hour,
+                    dialogState = TimeRangeViewState.DialogState.NONE
+                )
+            }
+            TimeRangeViewState.DialogState.NONE -> updateState {
+                copy(dialogState = TimeRangeViewState.DialogState.NONE)
+            }
+        }
+    }
+
+    private fun checkRangeError(startHour: Int?, endHour: Int?) {
+        if (startHour == null) {
+            showErrorState(isError = true)
+            return
+        }
+        val start: Int = startHour
+
+        if (endHour == null) {
+            showErrorState(isError = true)
+            return
+        }
+        val end: Int = endHour
+
+        if (start < end) {
+            showErrorState(isError = false)
+            return
+        }
+
+        showErrorState(isError = true, showSnackBar = true)
+    }
```

Comment:
```suggestion
private fun checkRangeError(startHour: Int?, endHour: Int?) {
    if (startHour == null || endHour == null) {
        showErrorState(isError = true)
        return
    } else if (startHour >= endHour) {
        showErrorState(isError = true, showSnackBar = true)
        return
    }
    
    showErrorState(isError = false)
}
```

이 부분 임시로 이렇게 하신건가요? 아니라면 이런식으로 코드를 리펙토링 할 수 있을거 같아요!

Reply:
suggest 방식에서도 start 와 end 모두 null 체크를 해주어서 else if 로 가면 둘다 nullable 하지 않게 캐스팅 되어서 비교가 가능한거 같던데 저렇게하면 에러가 나나요?

Reply:
조아용 스마트 캐스팅 되어서 비교가 가능하기에 참고해서 구현하시면 될 것 같슴다!

Reply:
오 위에 답글을 못 봤네요 ㅠㅠ
처음에 저런 방식으로 해뒀는데 스마트 캐스팅이 안 먹혀서 자꾸 빌드에 실패하더라고요
다시 해 보니 되는 걸 보니까 아마 버그였던가,, 그랬던 것 같은데.. (클린 후에 재빌드하면 괜찮았을지도.. 모른다는 생각이 드네용...)
저는 또 함수 파라미터로 전달받은 값은 스마트 캐스팅이 안 되게 되어 있다거나 그런 건 줄 알았어요 ㅠㅠㅋㅋㅋ쿠ㅜㅜㅜ 😅